pkg/pillar/cmd/volumemgr/sizemgmt.go

```diff
@@ -38,7 +38,15 @@ func getRemainingDiskSpace(ctxPtr *volumemgrContext) (uint64, error) {
 				iterVolumeStatus.Key(), iterVolumeStatus.State)
 			continue
 		}
-		totalDiskSize += iterVolumeStatus.MaxVolSize
+		cfg := lookupVolumeConfig(ctxPtr, iterVolumeStatus.Key())
+		if cfg != nil && !cfg.ReadOnly && !cfg.HasNoAppReferences {
+			totalDiskSize += iterVolumeStatus.MaxVolSize
+		} else {
+			// we have no config with this volume, so it will purged soon
+			// or it is ReadOnly and will not grow
+			// or it has no apps pointing onto it in new config
+			totalDiskSize += uint64(iterVolumeStatus.CurrentSize)
```

Comment:
I wonder if it would make sense to add some log.Noticef in here and do that with 4 unique messages for the 4 different cases. Wihout that it would be hard to verify that the calculation is correct until we run it on devices with insufficient space. Basically logging for testability.